### DISCUSSION TITLE
Help with exception : Unable to generate a URL for the named route "android-app://com.google.android.gm/" as such route does not exist

### DISCUSSION THREAD

**Discussion by sdespont** [symfony/symfony#50385] [Q&A] [answered]
I need help to understand this error occurring sometimes in my application.

I tried to log in from an Android phone without problem, so I really don't understand what is happening.

Here is the exception workflow

```
    "class": "Symfony\\Component\\Routing\\Exception\\RouteNotFoundException",
    "message": "Unable to generate a URL for the named route \"android-app://com.google.android.gm/\" as such route does not exist.",
    "code": 0,
    "file": ".../www/prod/site/vendor/symfony/routing/Generator/CompiledUrlGenerator.php:50",
    "trace": [
        ".../www/prod/site/vendor/symfony/routing/Router.php:235",
        ".../www/prod/site/vendor/symfony/security-http/HttpUtils.php:163",
        ".../www/prod/site/vendor/symfony/security-http/HttpUtils.php:70",
        ".../www/prod/site/vendor/symfony/security-http/Authentication/DefaultAuthenticationSuccessHandler.php:61",
```

And the link to the DefaultAuthenticationSuccessHandler class
https://github.com/symfony/symfony/blob/746c06b7869bdc081a2b61cbef33e94d70435483/src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationSuccessHandler.php#L55

Is there someone with an idea to help me out?

**Comment by MatTheCat** [comment]
This is because `HttpUtils::createRedirectResponse` treats `$path` as a route name if it does not start with `http` or `/`. You’ll have to pass a custom `success_handler` to your authentication listener configuration.

**Reply by sdespont** [reply]
OK, thanks for your answer. I don't understand how to reproduce this error occurring sometimes to the users. So.... I wil implement a custom listener and manage this specific case "android-app://com.google.android.gm/" manually.

**Reply by MatTheCat** [reply]
Is `android-app://com.google.android.gm/` a value in your configuration?

**Reply by sdespont** [reply]
No, absolutely not. It comes as referrer I suppose. For example, it occurs sometimes when user is validating his email (symfonycasts/verify-email-bundle), perhaps by clicking on the link from a native Gmail application (phone? tablet?) ? But I am not able to reproduce the error.

**Reply by sdespont** [reply]
Ok, I understand the problem now. In fact, after the user verifies its email, I automatically logged in the user. In the error, Monolog tells me that the referrer is "android-app://com.google.android.gm/" so it is part of the $request parameter and Symfony is trying to generate the corresponding route because the referrer is not starting by "http".

        // Authenticate user
        $authenticator->authenticateUser(
            $user,
            $this->formAuthenticator,
            $request
        );

Is there a way to set the referrer of $request ?

**Reply by MatTheCat** [reply]
I guess you should either configure [`use_referer`](https://symfony.com/doc/current/security/form_login.html#using-the-referring-url) to `false` (its default value), or create a custom success handler to implement whatever logic you want :thinking:

**Reply by sdespont** [reply]
No because I want to use the `use_referer` option, but `$request->headers->set('referer', null);` should do the job. I will try this PM

**Reply by MatTheCat** [reply]
Well if you set yourself the referrer it makes no sense using it.

**Reply by sdespont** [reply]
Yes, but I will set to NULL only in case of the referer is "android-app://com.google.android.gm/" to avoid the error in the email validation process

**Reply by MatTheCat** [reply, marked as answer]
Create your own success handler instead.

FYI I opened #50500 from this discussion.

**Reply by sdespont** [reply]
Thank you very much for your help !
